bundle/edu.gemini.pot/src/main/java/edu/gemini/pot/sp/memImpl/MemObservation.java

```diff
@@ -109,7 +109,13 @@ public int getObservationNumber() {
     }
 
     public void setObservationNumber(int number) {
-        _obsNumber = number;
+        getProgramWriteLock();
+        try {
+            _obsNumber = number;
+            ((ProgramData) getProgram().getDocumentData()).ensureMaxEqualToOrGreaterThan(number);
+        } finally {
+            returnProgramWriteLock();
+        }
```

Comment:
I agree.  It's kind of horrible for this to be public.  I was worried about temporarily inconsistent states.  If the observation will replace an existing observation with the same number, for example, then it is okay for it to be duplicated.  They just can't both be in the tree at the same time.  It's only when you add an observation or something that contains an observation, I think, that you can demand unique numbers.